core/src/main/scala/kafka/server/metadata/MetadataPartitions.scala

```diff
@@ -171,25 +202,38 @@ class MetadataPartitionsBuilder(val brokerId: Int,
     val prevPartition = newPartitionMap.put(partition.partitionIndex, partition)
     if (partition.isReplicaFor(brokerId)) {
       _localChanged.add(partition)
```

Comment:
Is it redundant to add this partition to `_localChanged` if the broker is already the replica of that partition?

Reply:
We depend on changes to the `_localChanged` set in order to receive new leader and isr state.